book/core-concepts/core-concepts.md

```diff
@@ -12,26 +12,37 @@ External   | Wallaroo                     Wallaroo|    External
 
-A Wallaroo application can have multiple interacting pipelines. For example, an app could have one pipeline that takes data and updates state based on that data, and a second pipeline that takes data, does computations against the current state in the system, and produces output based on the state and data. The first of these has no sink, whereas the second does.
+A Wallaroo application can have multiple sources. Each source might send data down a chain of computations. These chains can then be merged, say at a downstream state computation. For example:
+
+```
+External   | Wallaroo                    
+ System ──>|  Source ─> C1 ─> C2 -\
+   A       |                       \
+                                    \         Wallaroo|    External
+                                     -> C5 ->   Sink  |-->  System
+                                    /                 |       C
+External   | Wallaroo              / 
+ System ──>|  Source ─> C3 ─> C4 ─/ 
+   B       |                                   
+```
 
 ## Concepts
 
 * *State* -- Accumulated result of data stored over the course of time
-* *Computation* -- Code that transforms an input of some type `In` to
+* *Stateless Computation* -- Code that transforms an input of some type `In` to
 an output of some type `Out` (or optionally `None` if the input should be
 filtered out).
 * *State Computation* -- Code that takes an input type `In` and a state
 object of some type `State`, operates on that input and state (possibly
 making state updates), and optionally producing an output of some type `Out`.
-* *Source* -- Input point for data from external systems into an application.
-* *Sink* -- Output point from an application to external systems.
+* *Source* -- Ingress (input) point for data from external systems into a Wallaroo application.
+* *Sink* -- Egress (output) point from a Wallaroo application to external systems.
 * *Decoder* -- Code that transforms a stream of bytes from an external system
 into a series of application input types.
 * *Encoder* -- Code that transforms an application output type into bytes for
 sending to an external system.
-* *Pipeline* -- A sequence of computations and/or state computations originating
-from a source and optionally terminating in a sink.
-* *Application* -- A collection of pipelines.
+* *Pipeline* -- A sequence of computations and/or state computations originating from one or more sources and terminating in a sink.
+* *Stage* -- Any of the individual stages in a pipeline (each stage is either a source, a stateless computation, a state computation, or a sink).
```

Comment:
I'd rewrite this so as to not use "stage" as part of the definition of "stage".

Reply:
done